Plugins/Public/event/Main.cpp

```diff
@@ -703,6 +773,23 @@ void __stdcall GFGoodBuy_AFTER(struct SGFGoodBuyInfo const &gbi, unsigned int iC
 					}
 				}
 
+				// Is the event limited by Ship class?
+				if (i->second.isClassRestricted)
+				{
+					//PrintUserCmdText(iClientID, L"Checking ship class requirements....");
+					Archetype::Ship* TheShipArch = Archetype::GetShip(Players[iClientID].iShipArchetype);
+					wstring classname = shipclassnames.find(TheShipArch->iShipClass)->second;
+					
+					if (classname != i->second.allowedShipClass)
+					{
+						PrintUserCmdText(iClientID, L"DEBUG: Invalid ship class[%s], this Evnet is for [%s]'s", classname.c_str(), i->second.allowedShipClass.c_str());
+						return;
+					}
+				}
+				else {
+					i->second.allowedShipClass = stows("No Restrictions").c_str();
```

Comment:
Unnecessary c_str() and instead of converting string into a literal, just load it directly as wstring:
... = L"No Restrictions";